libtrip-notification/src/main/java/com/mapbox/navigation/trip/notification/MapboxTripNotification.kt

```diff
@@ -0,0 +1,29 @@
+package com.mapbox.navigation.trip.notification
+
+import com.mapbox.navigation.base.ExperimentalMapboxNavigationAPI
+
+/**
+ * This allows you to customize or extend the trip notification. Only one interceptor can be set
+ * at a time. To avoid memory leaks, remember to clear the interceptor when you no longer need to
+ * intercept notifications.
+ */
+@ExperimentalMapboxNavigationAPI
+object MapboxTripNotification {
```

Comment:
Debating deleting this object and passing it through MapboxNavigation

Reply:
I like this idea :+1: 

The notification is bound to `MapboxNavigation` lifecycle, so it doesn't need a static scope. `MapboxNavigation#setNotificationInterceptor` should work just fine.

@VysotskiVadim mentioned `NavigationOptions` above but I'm not a fan of having functional interfaces stored as options since they can easily leak this way. We are not the best at preventing this pattern but let's not make it worse.

Reply:
Trying to take the approach when passing through `MapboxNavigation`. `TripNotification` is an interface and it will cause a `Java` `SEMVER` to add something like a `TripNotification.setNotificationInterceptor`.

Using `NavigationOptions` is not ideal because Android Auto needs to be able to set/reset at runtime. So we need to have something like a `MapboxTripNotificationInterceptorProvider` when passed through there

Reply:
Passing through `NavigtationOptions` will also make it the `@MapboxModule` feature feel more loosely coupled. Because `MapboxNavigatation.setNotificationInterceptor` will not work when override your own module. And neither will an option such as `NavigtationOptions.notificationIntercepterProvider`

The `object MapboxTripNotification` will not for `@MapboxModule` either. So am not seeing a beautiful solution yet. We can also resolve one of these approaches with some documentation

🤔 .. Am going to explore yet another idea with `MapboxNavigationApp.getObserver`. That would require you use `MapboxNavigationApp`

Reply:
Hmm yeah, `MapboxNavigationApp` approach requires `MapboxNavigationApp`. 

`NavigationOptions` approach will avoid SEMVER thanks to the Builder pattern. We can document `@MapboxModule` issue in the comments for a `TripNotificationOptions`. I don't think we need a `MapboxTripNotificationInterceptorProvider` because the `interceptor` can decide when Android Auto is "active"